File: VB/E5059/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports DevExpress.XtraReports.UI
Imports NWindModel
Imports E5059.Reports

Namespace E5059.Controllers
	Public Class HomeController
		Inherits Controller
        Private Shared _dataContext As New NWindEntities()

		Public ReadOnly Property DataContext() As NWindEntities
			Get
				Return _dataContext
			End Get
		End Property

		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function DocumentViewerPartial() As ActionResult
			Dim report As XtraReport = New CategoryProductsReport()
			report.DataSource = _
				From category In DataContext.Categories _
				Select category
			Return PartialView("_DocumentViewerPartial", report)
		End Function

		Public Function DocumentViewerExport() As ActionResult
			Dim report As XtraReport = New CategoryProductsReport()
			report.DataSource = _
				From category In DataContext.Categories _
				Select category
			Return DocumentViewerExtension.ExportTo(report)
		End Function
	End Class
End Namespace

File: CS/E5059/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using DevExpress.XtraReports.UI;
using E5059.Models;
using E5059.Reports;

namespace E5059.Controllers {
    public class HomeController : Controller {
        static NWindEntities _dataContext = new NWindEntities();

        public NWindEntities DataContext {
            get {
                return _dataContext;
            }
        }

        public ActionResult Index() {
            return View();
        }

        public ActionResult DocumentViewerPartial() {
            XtraReport report = new CategoryProductsReport();
            report.DataSource = from category in DataContext.Categories
                                select category;
            return PartialView("_DocumentViewerPartial", report);
        }

        public ActionResult DocumentViewerExport() {
            XtraReport report = new CategoryProductsReport();
            report.DataSource = from category in DataContext.Categories
                                select category;
            return DocumentViewerExtension.ExportTo(report);
        }
    }
}